src/db.rs

```diff
@@ -171,3 +171,155 @@ impl From<Error> for Bytes {
         format!("{:?}", e).into()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    #[test]
+    fn test_manifest_new() {
+        let manifest = Manifest::new();
+        assert_eq!(
+            manifest,
+            Manifest {
+                logs: HashMap::new(),
+                itrs: HashMap::new(),
+            }
+        );
+    }
+    #[test]
+    fn test_manifest_add_log() {
+        let mut manifest = Manifest::new();
+        manifest.add_log("test".to_owned());
+        manifest.add_log("test2".to_owned());
+        manifest.add_log("test3".to_owned());
+        assert!(manifest.logs.contains_key("test"));
+        assert!(manifest.logs.contains_key("test2"));
+        assert!(manifest.logs.contains_key("test3"));
+        assert_eq!(manifest.logs.contains_key("test1"), false);
+
+        // This second add_log is here to make sure code does not panic
+        manifest.add_log("test".to_owned());
```

Comment:
What did you think of this behavior? Where adding a log that already exists is a no-op?  Part of me thought we should return a LogAlreadyExists error or something...but I really like Redis' ability to set up Lua scripts at the beginning and if they exist it's a no op. Makes writing db setup code nice since it can just happen everytime your program comes up instead of in a separate migration script.  Might be something to change though if it turns out to cause problems.

Reply:
I think you can keep it as a no-op but maybe return a 304 so it is obvious the api call isnt doing anything.

Reply:
Makes sense. Since TCP doesn't have status codes like HTTP does we can probably make a standard response that indicates a call was a no-op with the thing already being created.  Since that's not an _error_ per se, it also means we should probably think of a format for the responses to indicate error vs info vs data.